extensions/elation/actions/addHistories/addHistories.ts

```diff
@@ -0,0 +1,91 @@
+import { z } from 'zod'
+import { AxiosError } from 'axios'
+import { isNil } from 'lodash'
+
+import { type Action, Category, validate } from '@awell-health/extensions-core'
+import { SettingsValidationSchema, type settings } from '../../settings'
+import { makeAPIClient } from '../../client'
+import { FieldsValidationSchema, fields as elationFields } from './config'
+import { type HistoryTypeSchema } from '../../types/history'
+
+export const addHistories: Action<typeof elationFields, typeof settings> = {
+  key: 'addHistories',
+  category: Category.EHR_INTEGRATIONS,
+  title: 'Add Histories',
+  description: "Add history items on Elation's patient page",
+  fields: elationFields,
+  previewable: true,
+  onEvent: async ({ payload, onComplete, onError }): Promise<void> => {
+    try {
+      const { fields, settings } = validate({
+        schema: z.object({
+          fields: FieldsValidationSchema,
+          settings: SettingsValidationSchema,
+        }),
+        payload,
+      })
+      // we can execute up to 12 time so we want to capture if any of the fields failed
+      const failedFields: string[] = []
+
+      const api = makeAPIClient(settings)
+
+      // Get all field names except 'patientId'
+      const fieldNames = Object.keys(fields).filter(
+        (field) => field !== 'patientId'
+      )
+
+      // Attempt to add histories for each field
+      await Promise.all(
+        fieldNames.map(async (field) => {
+          const key = field as keyof typeof elationFields
+          const type = elationFields[key]?.label as z.infer<
+            typeof HistoryTypeSchema
+          >
+
+          try {
+            await api.addHistory({
+              type,
+              patient: fields.patientId,
+              text: field,
+            })
+          } catch (failedField) {
+            const errMessage =
+              failedField instanceof Error
+                ? failedField.message
+                : 'failed to add'
+            failedFields.push(`${type}: ${errMessage}`)
+          }
+        })
+      )
+
+      if (failedFields.length > 0) {
+        throw new Error(
+          `Failed to create histories for fields: ${failedFields.join(', ')}`
+        )
+      }
```

Comment:
there's a bit of a problem here. if 9/10 fields pass and one fails, we'll still be writing to the histories for 9/10 of them.
so, i think throwing an error here is not quite right.

do you know of a way we can guarantee all of the addHistory calls happen in a transaction?

Reply:
I did look into that - the API currently doesn't support adding multiple histories so we would have to manually delete them 1 by 1 (which can also throw errors in which case we end up in ugly mid-state).  I think this is a bit of a corner case and I've added this as a way to let the user know - hey this is missing. So I think for v1 that should be enough. LMK if you disagree.

Reply:
@bejoinka let me know if I'm okay to merge or if you want to chat more about the ^

Reply:
yeah, i had made this one open ended but indeed i don't think it's a good idea to "fail" an activity that successfully writes to the API. Is the expectation here that people will write one history at a time?